### PR TITLE
feat: added AccountWatcher as preinstalled snap and added to menu list

### PR DESCRIPTION
**Pull request by k-g-j**
## **Description**
This PR adds a new button in the Add Account Menu to direct users to the Account Watcher Homepage. The new feature is hidden behind an Experimental Feature toggle that is off by default for a stealthy launch. 

Key changes include:
1. Added a new menu item in the Add Account Menu for watching Ethereum accounts.
2. Implemented an Experimental Feature toggle for "Watch Ethereum Accounts (Beta)".
3. Added metric tracking for the toggle state.
4. Ensured the toggle is disabled by default.

Note: flow and code changes based on [support creation of Bitcoin testnet accounts](https://github.com/MetaMask/metamask-extension/pull/25772)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26402?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/accounts-planning/issues/468#issue-2331781414

## **Manual testing steps**
1. Go to Settings > Experimental
2. Locate the "Watch Ethereum Accounts (Beta)" toggle
3. Enable the toggle
4. Navigate to the Add Account Menu
6. Verify that the new "Watch Ethereum account (Beta)" option is visible
7. Click on the new option and confirm it redirects to the Account Watcher Homepage
8. Watch an Ethereum account or ENS (mainnet)
9. Navigate back to Account Watcher Homepage and press the back arrow in the modal header. It _should_ redirect you to the default wallet page. 
10. Disable the toggle in Settings > Experimental
11. Verify that the "Watch Ethereum Account" option is no longer visible in the Add Account Menu

## **Screenshots/Recordings**

Screenshot 2024-08-13 at 5 30 53 PM

Screenshot 2024-08-14 at 10 07 13 AM

https://github.com/user-attachments/assets/6ea131db-f733-4258-83e7-7569bcffd062

## **Pre-merge author checklist**
- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**
- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by k-g-j** [comment]
@SocketSecurity ignore npm/@metamask/account-watcher@4.0.2

**Comment by k-g-j** [comment]
@SocketSecurity ignore npm/@metamask/account-watcher@4.0.3

File: ui/components/multichain/account-list-menu/account-list-menu.tsx [hunk 1]
```diff
@@ -218,6 +225,23 @@ export const AccountListMenu = ({
   ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
   const addSnapAccountEnabled = useSelector(getIsAddSnapAccountEnabled);
   ///: END:ONLY_INCLUDE_IF
+  const isAddWatchEthereumAccountEnabled = useSelector(
+    getIsWatchEthereumAccountEnabled,
+  );
+  const handleAddWatchAccount = async () => {
```
Comment (owencraston) [inline ui/components/multichain/account-list-menu/account-list-menu.tsx:231]: useCallback

File: ui/components/multichain/account-list-menu/account-list-menu.tsx [hunk 2]
```diff
@@ -520,6 +544,19 @@ export const AccountListMenu = ({
...
               ///: END:ONLY_INCLUDE_IF
             }
+            {isAddWatchEthereumAccountEnabled && (
+              <Box marginTop={4}>
+                <ButtonLink
+                  disabled={!isAddWatchEthereumAccountEnabled}
+                  size={ButtonLinkSize.Sm}
+                  startIconName={IconName.Eye}
+                  onClick={handleAddWatchAccount}
+                  data-testid="multichain-account-menu-popover-add-watch-only-account"
```
Comment (owencraston) [inline ui/components/multichain/account-list-menu/account-list-menu.tsx:554]: should the test ids be set to constants somewhere?

File: ui/pages/settings/experimental-tab/experimental-tab.component.tsx [hunk 3]
```diff
@@ -349,6 +385,7 @@ export default class ExperimentalTab extends PureComponent<ExperimentalTabProps>
           this.renderKeyringSnapsToggle()
           ///: END:ONLY_INCLUDE_IF
         }
+        {this.renderWatchAccountToggle()}
```
Comment (owencraston) [inline ui/pages/settings/experimental-tab/experimental-tab.component.tsx:388]: are we not putting this feature behind a code fence first?

**Comment by owencraston** [comment]
Tested and everything looks great. Here are a few comments


https://github.com/user-attachments/assets/c632e560-db4c-435c-988f-698b34c2977e


## Flows tested
- no watch only account option in add account menu when toggle is off (default case) ✅ 
- turning on the watch only account experimental toggle in settings enables watch only accounts in the add account menu ✅ 
- watching an account with an ENS name on a testnet throws an error ✅ 
- watch an account with an ENS name on mainnet works as exptected ✅ 
- newly watched account becomes the selected account ✅ 
- send/bridge/swap flows are disabled
- cannot watch same account twice ✅ 
- turning off  watch only in experimental does not remove existing watch only accounts ✅ 
- account list shows 'Account watcher (Beta)' pill ✅ 
- experimental settings content matches spec ✅ 
- feedback form links to correct form ✅ 

## Questions
- dapp connections still work and  result in some weird behaviour/failed transactions 

Screenshot 2024-08-15 at 3 27 35 PM

- designs differ slightly than what was specifie
Screenshot 2024-08-15 at 3 15 03 PM

File: ui/pages/settings/experimental-tab/experimental-tab.component.tsx (hunk 3 above)
Comment (k-g-j) [reply ui/pages/settings/experimental-tab/experimental-tab.component.tsx:388]: @AlexJupiter said we will put it behind an experimental toggle, but no code fencing required 😌

File: ui/components/multichain/account-list-menu/account-list-menu.tsx (hunk 2 above)
Comment (k-g-j) [reply ui/components/multichain/account-list-menu/account-list-menu.tsx:554]: I don't think there is a constants file in the extension for test IDs, but I will do some digging!

**Comment by k-g-j** [comment]
> designs differ slightly than what was specified

Yes, the Snap UI is limited by what is possible in their component system

**Comment by k-g-j** [comment]
> dapp connections still work and result in some weird behaviour/failed transactions

hmmmm this is a good point! - we should see if there is a way to disabled dapps from connecting to watch accounts  . . .

File: ui/components/multichain/account-list-menu/account-list-menu.tsx (hunk 1 above)
Comment (k-g-j) [reply ui/components/multichain/account-list-menu/account-list-menu.tsx]: done! [4d963aa](https://github.com/MetaMask/metamask-extension/pull/26402/commits/4d963aaecc95b757cb852e8849644ea143da134a)

**Comment by k-g-j** [comment]
@SocketSecurity ignore npm/@metamask/account-watcher@4.1.0

**Comment by k-g-j** [comment]
@metamaskbot update-policies

File: ui/components/multichain/account-list-menu/account-list-menu.tsx (hunk 2 above)
Comment (owencraston) [reply ui/components/multichain/account-list-menu/account-list-menu.tsx:554]: fair enough.

**Review by owencraston — APPROVED** [review]
Retested with the prod builds and everything looks good to me. Good work!

https://github.com/user-attachments/assets/6af7cc2c-b914-456d-bc4b-a1246cb88413

https://github.com/user-attachments/assets/adbefcb8-c146-49ce-b6fa-1adacb9ea0f9

**Review by jpuri — APPROVED** [review]
Changes in code owned by confirmations team look good 👍

**Review by matthewwalsh0 — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**